gapir/client/client.go

```diff
@@ -28,88 +33,201 @@ import (
 
 type tyLaunchArgsKey string
 
-// LaunchArgsKey is the bind device property key used to control the command
-// line arguments when launching GAPIR. The property must be of type []string.
-const LaunchArgsKey tyLaunchArgsKey = "gapir-launch-args"
+const (
+	// LaunchArgsKey is the bind device property key used to control the command
+	// line arguments when launching GAPIR. The property must be of type []string.
+	LaunchArgsKey     tyLaunchArgsKey = "gapir-launch-args"
+	connectTimeout                    = time.Second * 10
+	heartbeatInterval                 = time.Millisecond * 500
+)
+
+type clientInfo struct {
+	device               bind.Device
+	arch                 device.Architecture
+	abi                  *device.ABI
+	deviceConnectionInfo deviceConnectionInfo
+	connection           gapir.Connection
+	bgConnection         *backgroundConnection
+}
+
+type deviceArch struct {
+	device bind.Device
+	arch   device.Architecture
+}
+
+// ConnectionKey is used by manager to obtain a connection
+type ConnectionKey deviceArch
 
-// Client is interface used to connect to GAPIR instances on devices.
+// Client handles connections to GAPIR instances on devices.
+// A single Client can handle multiple connections.
 type Client struct {
-	mutex    sync.Mutex
-	sessions map[deviceArch]*session
+	// Mutex is needed due to reconnect may happen in another thread
```

Comment:
You cannot say "due to ... may happen", but rather: `// Mutex is needed because reconnect() may happen in another thread`